src/event_batch.rs

```diff
@@ -30,4 +36,100 @@ impl EventBatch {
             data: json!(self.events),
         }
     }
+
+    /// Updates the events `stm` field in batch with the current time.
+    pub fn update_event_stm(&mut self) -> Result<(), Error> {
+        let since_the_epoch =
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .map_err(|e: SystemTimeError| {
+                    Error::BuilderError(format!("Failed to get current time: {}", e.to_string()))
+                })?;
+
+        for event in self.events.iter_mut() {
+            event.stm = since_the_epoch.as_millis().to_string();
+        }
+
+        Ok(())
+    }
+
+    ///
+    pub fn update_delay(&mut self) {
+        let max_event_delay_time = Duration::from_secs(10);
```

Comment:
Shouldn't this be in minutes? The Java tracker has 10 minutes, so 600000 seconds.